File: python/20-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []

        for char in s:
            if char == '{' or char == '[' or char == '(':
                stack.append(char)
            elif (
                (char == ']' and len(stack) > 0 and stack[-1] == '[') 
                or (char == '}' and len(stack) > 0 and stack[-1] == '{') 
                or (char == ')' and len(stack) > 0 and stack[-1] == '(')
            ):
                stack.pop()
            else:
                return False

        return len(stack) == 0        

File: javascript/20-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
  let stack = [];
  let obj = {
      ")": "(",
      "]": "[",
      "}": "{",
  }

  for(let i = 0; i < s.length; i++) {
      let curr = s.charAt(i);

      if (curr === "(" || curr === "{" || curr === "[") {
          stack.push(curr);
      } else {
          if (stack.length === 0 || obj[curr] !== stack[stack.length - 1]) {
              return false;
          } else {
              stack.pop();
          }
      }
  }

  return stack.length === 0;
};